woocommerce/wc-admin
# Move Download table logic inside the TableCard component

While working on #621, I noticed we were repeating the same code to download tables as CSV's in every component that contains a downloadable table:
```ES6
downloadCSVFile(
	generateCSVFileName( 'revenue', query ),
	generateCSVDataFromTable( headers, rows )
);
```

Given that the process to download a table as a CSV is the same no matter what the table is, I think it makes sense to move this logic inside the `<TableCard>` component.

This PR does that and adds a `downloadable` prop to `TableCard` that will display the download button for the table. I didn't remove the `onClickDownload` prop, beucase we might want to use it in the future when implementing big data sets download via an API request.

### Screenshots

_(this PR shouldn't produce any visible change)_

### Detailed test instructions:

 - Go to any page with a table that can be downloaded (`/wp-admin/admin.php?page=wc-admin#/analytics/revenue` and `/wp-admin/admin.php?page=wc-admin#/analytics/orders`).
- Download the table and verify the process works properly.
